weasel/_util/general.py

```diff
@@ -0,0 +1,568 @@
+import hashlib
```

Comment:
It would be helpful to know which of these methods have been copied verbatim from `spaCy` and which are new or changed.

Reply:
That's a good point. Unfortunately we weren't keeping track of that as this was written - it may be a little hard to untangle things now, but I'll see what I can do.

Reply:
I have looked into this and unfortunately I can't find a good way to reduce the size of diffs on Github, which would be the easiest way to do this. 

Since these functions were generally undertested anyway, and it's not *that* much code, I think it's best to just review everything as though it were fresh.